tests/dialects/test_stablehlo.py

```diff
@@ -0,0 +1,23 @@
+from xdsl.dialects.builtin import AnyTensorType, IntegerType, TensorType
+from xdsl.dialects.stablehlo import (
+    AddOp,
+    _abstract_operation_class_factory,  # pyright: ignore[reportPrivateUsage]
+)
+from xdsl.irdl import VarConstraint, base
+from xdsl.utils.test_value import TestSSAValue
+
+
+def test_abstract_factory():
+    unop, binop = _abstract_operation_class_factory(
+        "", VarConstraint("T", base(AnyTensorType))
+    )
+    unop.operand
+    binop.lhs
+    binop.rhs
+
+
+def test_impact_of_abstract_factory_in_ops_that_uses_base_classes():
+    a = TestSSAValue(TensorType(IntegerType(32), []))
+    addop = AddOp(a, a)
+    addop.lhs
+    addop.rhs
```

Comment:
Does this work?
`from pyright import reveal_type`
```suggestion
    reveal_type(addop.rhs, expected_text="TensorType")
```

Reply:
information: Type of "addop.rhs" is "SSAValue". 🤔

Reply:
OK this is fantastic, how about `addop.T`? Sounds like it's able to infer the types through the function call, which is what I was worried about.

Reply:
I think actually for both tests, it would be nice to add everything that we might want to have the type for as users (also the new type should be richer than `SSAValue` if you update main :) )